src/fsharp/FSharp.Core/array.fs

```diff
@@ -478,6 +478,17 @@ namespace Microsoft.FSharp.Collections
         [<CompiledName("Where")>]
         let where f (array: _[]) = filter f array
 
+        [<CompiledName("Except")>]
+        let except (array1:'T[]) (array2:'T[]) =
+            checkNonNull "array1" array1
+            checkNonNull "array2" array2
+
+            if array1.Length = 0 then
```

Comment:
If array2 is empty and array1 is not, then the function returns the distinct elements of array1